widget/menu_item.go

```diff
@@ -78,8 +97,20 @@ func (i *menuItem) MinSize() fyne.Size {
 // MouseIn changes the item to be hovered and shows the submenu if the item has one.
 // The submenu of any sibling of the item will be hidden.
 // Implements: desktop.Hoverable
-func (i *menuItem) MouseIn(*desktop.MouseEvent) {
+func (i *menuItem) MouseIn(e *desktop.MouseEvent) {
 	i.hovered = true
+	if i.Child() != nil {
+		i.Child().Defocus()
+		if e == nil && i.Child().Items[0] != nil {
```

Comment:
This seems wrong - you are sending "nil" for keyboard events but simulating mouse events - gotta be a cleaner way.

Reply:
I would also like a cleaner way. But I can not find any, except by duplicating a lot of code.

Reply:
Why does the code need to be duplicated? Can’t common code be extracted into separate methods that are nicely named to make it clearer?